[main](backport #44749) Add warning about Filestream only ingesting files >= 1024 bytes. (#44784)
elastic/beats: docs/reference/filebeat/filebeat-input-filestream.md

# filestream input [filebeat-input-filestream]

::::{important}
By default, Filestream will only start ingesting files **when
their size is >= 1024 bytes**. To change this behaviour you can
configure a different length for the fingerprint by setting

> [`prospector.scanner.fingerprint.length`](#filebeat-input-filestream-scan-fingerprint).

Reviewer: I like the re-wording, but I'd keep the bold emphasis. This whole change came from people having trouble ingesting small files because the fact that Filestream won't ingest files <1024 bytes by default was not visible enough.

What do you think about keeping the bold?

::::

Use the `filestream` input to read lines from active log files. It is the new, improved alternative to the `log` input. It comes with various improvements to the existing input:

> * The default behaviour is to identify files based on their contents using [`fingerprint`](#filebeat-input-filestream-file-identity-fingerprint) [`file_identity`](#filebeat-input-filestream-file-identity). This solves data duplication caused by inode reuse.
> * Checking of `close.on_state_change.*` options happens out of band. Thus, if an output is blocked, Filebeat can close the reader and avoid keeping too many files open.
> * Detailed metrics are available for all files that match the `paths` configuration regardless of the `harvester_limit`. This way, you can keep track of all files, even ones that are not actively read.
> * The order of `parsers` is configurable. So it is possible to parse JSON lines and then aggregate the contents into a multiline event.
> * Some position updates and metadata changes no longer depend on the publishing pipeline. If the pipeline is blocked some changes are still applied to the registry.
> * Only the most recent updates are serialized to the registry. In contrast, the `log` input has to serialize the complete registry on each ACK from the outputs. This makes the registry updates much quicker with this input.
> * The input ensures that only offsets updates are written to the registry append only log. The `log` writes the complete file state.
> * Stale entries can be removed from the registry, even if there is no active input.

Reviewer 2: ```suggestion
* The default behavior is to identify files based on their contents using [`fingerprint`](#filebeat-input-filestream-file-identity-fingerprint) [`file_identity`](#filebeat-input-filestream-file-identity). This solves data duplication caused by inode reuse.
* Validation of `close.on_state_change.*` options happens out of band. If an output is blocked, Filebeat can close the reader and avoid keeping too many files open.
* Detailed metrics are available for all files that match the `paths` configuration regardless of the `harvester_limit`. This way, you can keep track of all files, even ones that are not actively read.
* The order of `parsers` is configurable. You can parse JSON lines and then aggregate the contents into a multiline event.
* Some position updates and metadata changes no longer depend on the publishing pipeline. If the pipeline is blocked, some changes are still applied to the registry.
* Only the most recent updates are serialized to the registry. In contrast, the `log` input has to serialize the complete registry on each ACK from the outputs. This makes the registry updates much quicker with this input.
* The input ensures that only offsets updates are written to the registry append only log. The `log` writes the complete file state.
* Stale entries can be removed from the registry, even if there is no active input.
```
